app/src/main/java/nl/tudelft/cs4160/trustchain_android/peer/PeerHandler.java

```diff
@@ -236,9 +237,11 @@ public synchronized Peer getOrMakePeer(InetSocketAddress address, PublicKeyPair
             if (peer.getAddress().equals(address)) {
                 if (publicKeyPair != null) peer.setPublicKeyPair(publicKeyPair);
                 peer.setName(name);
+                removeDuplicates();
```

Comment:
This might cause ConcurrentModificationExceptions if I'm right

Reply:
No, removeDuplicates is syncronized.

Reply:
Still, if removeDuplicates() removes a peer there, the running loop will fail

Reply:
That does not matter, removing while iterating like this will cause exceptions run this example:

```
    @Test
    public void testCrash() {
        List<Integer> list = new ArrayList<Integer>(2);
        list.add(2);
        list.add(5);
        for(Integer x : list) {
            if(x == 5) {
//                list.remove(x);
                remove(list, x);
            }
        }
    }
    public synchronized void remove(List<Integer> list, Integer toRemove) {
        for(int i=0; i< list.size(); i++) {
            if(toRemove.equals(list.get(i))) {
                list.remove(toRemove);
            }
        }
    }
```

Reply:
It always returns right after executing that method, so the loop is not continuing.

Reply:
So it is bug prone, but not a bug yet. :P